pkg/solana/chainwriter/chain_writer_test.go

```diff
@@ -0,0 +1,690 @@
+										Seeds: []chainwriter.Lookup{
+											// extract seed2 for PDA lookup
+											chainwriter.AccountLookup{Name: "seed2", Location: "seed2"},
+										},
+										IsSigner:   false,
+										IsWritable: false,
+										InternalField: chainwriter.InternalField{
+											Type:     reflect.TypeOf(chainwriter.DataAccount{}),
+											Location: "LookupTable",
+										},
+									},
+								},
+							},
+							StaticLookupTables: []string{staticLookupTablePubkey.String()},
+						},
+						Accounts: []chainwriter.Lookup{
+							chainwriter.AccountConstant{
+								Name:       "Constant",
+								Address:    account1.String(),
+								IsSigner:   false,
+								IsWritable: false,
+							},
+							chainwriter.AccountLookup{
+								Name:       "LookupTable",
+								Location:   "lookup_table",
+								IsSigner:   false,
+								IsWritable: false,
+							},
+							chainwriter.PDALookups{
+								Name:      "DataAccountPDA",
+								PublicKey: chainwriter.AccountConstant{Name: "WriteTest", Address: solana.SystemProgramID.String()},
+								Seeds: []chainwriter.Lookup{
+									// extract seed1 for PDA lookup
+									chainwriter.AccountLookup{Name: "seed1", Location: "seed1"},
+								},
+								IsSigner:   false,
+								IsWritable: false,
+								// Just get the address of the account, nothing internal.
+								InternalField: chainwriter.InternalField{},
+							},
+							chainwriter.AccountsFromLookupTable{
+								LookupTableName: "DerivedTable",
+								IncludeIndexes:  []int{0},
+							},
+						},
+					},
+				},
+				IDL: testContractIDLJson,
+			},
+		},
+	}
+
+	// initialize chain writer
+	cw, err := chainwriter.NewSolanaChainWriterService(testutils.NewNullLogger(), rw, txm, ge, cwConfig)
+	require.NoError(t, err)
+
+	t.Run("fails with invalid ABI", func(t *testing.T) {
+		invalidCWConfig := chainwriter.ChainWriterConfig{
+			Programs: map[string]chainwriter.ProgramConfig{
+				"invalid_program": {
+					Methods: map[string]chainwriter.MethodConfig{
+						"invalid": {
+							ChainSpecificName: "invalid",
+						},
+					},
+					IDL: "",
+				},
+			},
+		}
+
+		_, err := chainwriter.NewSolanaChainWriterService(testutils.NewNullLogger(), rw, txm, ge, invalidCWConfig)
+		require.Error(t, err)
+	})
+
+	t.Run("fails to encode payload if args with missing values provided", func(t *testing.T) {
+		txID := uuid.NewString()
+		args := map[string]interface{}{}
+		submitErr := cw.SubmitTransaction(ctx, "contractReaderInterface", "initializeLookupTable", args, txID, programID.String(), nil, nil)
+		require.Error(t, submitErr)
+	})
+
+	t.Run("fails if invalid contract name provided", func(t *testing.T) {
+		txID := uuid.NewString()
+		args := map[string]interface{}{}
+		submitErr := cw.SubmitTransaction(ctx, "contractReaderInterface", "initializeLookupTable", args, txID, programID.String(), nil, nil)
```

Comment:
Huh seems like tests are passing but this might be picking up on a different error. I think we'll need to pass in a bad contract name here for this invalid case.
```suggestion
		submitErr := cw.SubmitTransaction(ctx, "badContract", "initializeLookupTable", args, txID, programID.String(), nil, nil)
```